### PR TITLE
Use Rollup to bundle files

### PR DESCRIPTION
**Pull request by Turbo87**
This PR is using `broccoli-rollup` to treeshake and bundle files in the `ember-test-helpers`, `mocha` and `ember-mocha` modules.

Note that this is irrelevant for `ember-cli-mocha`, which imports the ES6 files instead.

### CODE REVIEW

**Comment by Turbo87** [comment]
it looks like this is blocked on #118 since Rollup requires a native `Promise` class which is only available starting at Node.js v0.12

**Comment by Turbo87** [comment]
@rwjblue any comments?

File: ember-cli-build.js [hunk 1]
```diff
@@ -47,7 +40,47 @@ module.exports = function(defaults) {
...
+  var emberTestHelpersPath = path.dirname(resolve.sync('ember-test-helpers'));
+
+  var emberTestHelpersRollup = new Rollup(emberTestHelpersPath, {
+    rollup: {
+      entry: 'ember-test-helpers.js',
+      dest: 'ember-test-helpers.js',
+      external: ['ember'],
+      plugins: [
+        alias({
+          'ember-test-helpers/': './ember-test-helpers/',
```
Comment (rwjblue) [inline ember-cli-build.js]: Not sure why we need this alias, who is using `./ember-test-helpers` like this?

File: tests/mocha-module-test.js [hunk 2]
```diff
@@ -1,34 +0,0 @@
-import { createModule } from 'ember-mocha/mocha-module';
```
Comment (rwjblue) [inline tests/mocha-module-test.js]: Do we not need this file any more?

Comment (Turbo87) [inline tests/mocha-module-test.js]: as mentioned in the commit message:
> These things are tested in the other tests too and `createModule` is not public API

**Pull request merged**